### ISSUE TITLE
Wrong result when using POSITION() function with only bind values from JDBC

### ISSUE THREAD

**Issue by lukaseder** [duckdb/duckdb#10695]
### What happens?

When using the `POSITION()` function with only bind value arguments, the result is wrong

### To Reproduce

```java
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;

import org.duckdb.DuckDBConnection;


DuckDBConnection connection = (DuckDBConnection) DriverManager.getConnection("jdbc:duckdb:");
try (PreparedStatement s = connection.prepareStatement("""
    select
        position(? in ?),
        position(? in 'abc'),
        position('b' in ?),
        position('b' in 'abc')
    """)) {
    s.setString(1, "b");
    s.setString(2, "abc");
    s.setString(3, "b");
    s.setString(4, "abc");

    try (ResultSet rs = s.executeQuery()) {
        while (rs.next()) {
            System.out.println(rs.getInt(1));
            System.out.println(rs.getInt(2));
            System.out.println(rs.getInt(3));
            System.out.println(rs.getInt(4));
        }
    }
}
```

This prints:

```
0
2
2
2
```

But it should print:

```
2
2
2
2
```

### OS:

Microsoft Windows [Version 10.0.22631.3007]

### DuckDB Version:

0.10.0

### DuckDB Client:

jdbc

### Full Name:

Lukas Eder

### Affiliation:

Data Geekery

### Have you tried this on the latest [nightly build](https://duckdb.org/docs/installation/?version=main)?

I have tested with a nightly build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by tlm365** [comment]
I looked at the code, this seems to be a bug. The `position(search_string in string)` function under the hood is `instr(string, search_string)` function ([code_ref](https://github.com/duckdb/duckdb/blob/d79d225fbb1e796ac80d3f15db0a56035f8e0b5c/src/include/duckdb/core_functions/scalar/string_functions.hpp#L147-L166)) ([docs](https://duckdb.org/docs/sql/functions/char.html)), but their parameters are reversed. This leads to the fact that when using `?` place holder in `PreparedStatement`, parameters will not be passed as expected.
@lukaseder In this case I recommend you use the `instr`/`strpos` function instead of `position` until this issue is resolved, or if you still want to use the `position` code that looks like this should work...

```java
DuckDBConnection connection = (DuckDBConnection) DriverManager.getConnection("jdbc:duckdb:");
try (PreparedStatement s = connection.prepareStatement("select position(? in ?)")) {
    s.setString(1, "abc");
    s.setString(2, "b");
    
    try (ResultSet rs = s.executeQuery()) {
        while (rs.next()) {
            System.out.println(rs.getInt(1));
        }
    }
}
catch (SQLException e) {
    throw new RuntimeException(e);
}
```
@szarnyasg Can you take a look at this problem? I'm willing to discuss and resolve this issue.

**Comment by Mytherin** [comment]
Thanks for the report! I've pushed a fix in #11314. Indeed as found by [tlm365](https://github.com/tlm365) the order of the arguments was inverted by the Postgres parser which caused the incremental `?` prepared statement parameters to be bound incorrectly. This would work correctly with explicitly numbered parameters (e.g. `SELECT POSITION($1 IN $2)`) which is the only kind that Postgres supports - hence why this is not a problem for them.
